dlc/src/util.rs

```diff
@@ -256,3 +256,31 @@ pub fn validate_fee_rate(fee_rate_per_vb: u64) -> Result<(), Error> {
 
     Ok(())
 }
+
+/// Computes the total weight of the given inputs and change.
+pub fn get_inputs_and_change_weight(
```

Comment:
Wouldn't it be better to use InputWeightPrediction https://github.com/rust-bitcoin/rust-bitcoin/blob/master/bitcoin/src/blockdata/transaction.rs#L1473 if this is a yet to be spent input?

Reply:
Sure, that looks even better. Thanks for the tip!

Reply:
Sure, I believe this code pre-dates the introduction of these utilities.

Reply:
IIUC `InputPredictionWeight` is meant to be used with the `predict_weight` method, but that computes the weight of an entire transaction, not just the input. As for the `weight` methods on `TxIn`, it would work only if the witness was already set (tx signed), so it doesn't seem very helpful either. Or do you guys have an idea on how to make it work?

Reply:
the comment here: https://github.com/rust-bitcoin/rust-bitcoin/blob/master/bitcoin/src/blockdata/transaction.rs#L1483 is "/// Weight prediction of an individual input.".  Can't you just provide `script_size` and `witness_size` to build the struct and then call `weight` here https://github.com/rust-bitcoin/rust-bitcoin/blob/master/bitcoin/src/blockdata/transaction.rs#L1627..

Reply:
I saw this method but IFAICT this computes only the weight of the script and witness, not the transaction input base weight, or am I mis-reading it?

Reply:
Maybe `InputWeightPrediction` does not work the way we want it to. But you can definitely call `segwit_weight` (or `legacy_weight) on `TxIn` and it will take into account the base weight.

Reply:
But not the witness if it's not filled AFAICT. I think the best if we really want to use some rust-bitcoin utilities is using `base_size` on `TxIn` but not really convinced it's worth the refactor.

Reply:
Yeah, if you don't think it's straightforward let's not do it.

Reply:
> I saw this method but IFAICT this computes only the weight of the script and witness, not the transaction input base weight, or am I mis-reading it?

If TxIn::BASE_WEIGHT is public, would that work then?  It sounds like what you're saying is you would need to do something like:

```rust
InputWeightPrediction.weight() + TxIn::BASE_WEIGHT`
```

I was thinking of also using `InputWeightPrediction` for coin-selection although maybe there's some alterations that are needed..

Reply:
I think it would be good to have all these constant values public for sure, then I could probably get rid of a lot of the ones that are in rust-dlc. But I'm not sure it'd be worth building an `InputWeightPrediction` struct to essentially perform an addition and a multiplication.

Reply:
> I think it would be good to have all these constant values public for sure, then I could probably get rid of a lot of the ones that are in rust-dlc.

I put in a PR a while back https://github.com/rust-bitcoin/rust-bitcoin/pull/2217 to make consts public but the feedback I received was that it's preferable to expose an API for what's needed instead of making a bunch of pub consts.  Is there some API interface that we could add instead?

> But I'm not sure it'd be worth building an InputWeightPrediction struct to essentially perform an addition and a multiplication.

It looks like this `get_inputs_and_change_weight` only computes the weight of `p2sh-p2w*`?  If this took an array of `InputWeightPrediction` instead `(Script, usize)` then this would work for different kinds of inputs like taproot for example..